src/components/Map.js

```diff
@@ -84,6 +91,8 @@ class Map extends Component {
       google.maps.event.addListener(marker, 'click', function (){ //eslint-disable-line
         if (currentSelectedMarker) {
           currentSelectedMarker.message.close();
+        }  else {
+          context.handleClick( pin.event );
```

Comment:
Can you call context.props.changeCurrEvent( pin.event ) directly to cut out a few lines of code?